Source/Blazorise.Tailwind/Config.cs

```diff
@@ -0,0 +1,84 @@
+﻿#region Using directives
+using System;
+using System.Collections.Generic;
+using Blazorise.Modules;
+using Microsoft.Extensions.DependencyInjection;
+#endregion
+
+namespace Blazorise.Tailwind;
+
+public static class Config
+{
+    /// <summary>
+    /// Adds a bootstrap providers and component mappings.
+    /// </summary>
+    /// <param name="serviceCollection"></param>
+    /// <returns></returns>
+    public static IServiceCollection AddTailwindProviders( this IServiceCollection serviceCollection, Action<IClassProvider> configureClassProvider = null )
+    {
+        var classProvider = new TailwindClassProvider();
+
+        configureClassProvider?.Invoke( classProvider );
+
+        serviceCollection.AddSingleton<IClassProvider>( classProvider );
+        serviceCollection.AddSingleton<IStyleProvider, TailwindStyleProvider>();
+        serviceCollection.AddSingleton<IBehaviourProvider, TailwindBehaviourProvider>();
+        serviceCollection.AddScoped<IThemeGenerator, TailwindThemeGenerator>();
+
+        serviceCollection.AddTailwindComponents();
+
+        serviceCollection.AddScoped<IJSModalModule, Modules.TailwindJSModalModule>();
+        serviceCollection.AddScoped<IJSTooltipModule, Modules.TailwindJSTooltipModule>();
+
+        return serviceCollection;
+    }
+
+    public static IServiceCollection AddTailwindComponents( this IServiceCollection serviceCollection )
+    {
+        foreach ( var mapping in ComponentMap )
+        {
+            serviceCollection.AddTransient( mapping.Key, mapping.Value );
+        }
+
+        return serviceCollection;
+    }
+
+    public static IDictionary<Type, Type> ComponentMap => new Dictionary<Type, Type>
```

Comment:
Actually, there is. We exposed ComponentMap some time ago so that users could add their own implementations to the DI. It was a PR from some member if I recall it.

Reply:
I see. Still it's more of a cookie cutter thing... People can overrid DI at will as far as I know. This just adds a little bit of memory imprint for maybe no reason since it's only used on startup... Either way it's fine I guess.